### PR TITLE
add error handling for embed

### PR DESCRIPTION
**Pull request by vellip**

### CODE REVIEW

File: meinberlin/assets/scss/style.scss [hunk 1]
```diff
@@ -26,7 +26,9 @@
 @import 'components/commenting';
 @import 'components/comments';
 @import 'components/dropdown';
+@import "components/embed-confirm";
```
Comment (xi) [inline meinberlin/assets/scss/style.scss]: merge error?

File: meinberlin/static/embed.js [hunk 2]
```diff
@@ -114,6 +114,35 @@ $(document).ready(function () {
...
+      default:
+        text = django.gettext('Something went wrong!')
+        break
+    }
+
+    $error.text(text)
+    $error.append($close)
+    $error.prependTo($('#embed-status'))
+
+    $close.on('click', removeMessage)
```
Comment (xi) [inline meinberlin/static/embed.js]: I would prefer if `$error` was completely clickable.

File: meinberlin/assets/scss/components/_embed-status.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
+.embed-status {
+    margin-top: 1em;
```
Comment (xi) [inline meinberlin/assets/scss/components/_embed-status.scss]: IMHO this looks strange

File: meinberlin/assets/scss/components/_embed-status.scss [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+.embed-status__message--error {
+    background-color: $brand-danger;
+    color: contrast-color($brand-danger);
+}
+
+.embed-status__close {
+    float: right;
+    color: contrast-color($brand-danger);
+}
```
Comment (xi) [inline meinberlin/assets/scss/components/_embed-status.scss]: I would prefer to reuse `alert` skin and only use different layout.

File: meinberlin/static/embed.js [hunk 5]
```diff
@@ -114,6 +114,35 @@ $(document).ready(function () {
     }
   }, false)
 
+  $(document).ajaxError(function (event, jqxhr) {
+    var text
+    var $error = $('<p class="embed-status__message embed-status__message--error"></p>')
+    var $close = $('<button class="embed-status__close"><i class="fa fa-times"></i></button>')
```
Comment (xi) [inline meinberlin/static/embed.js]: `<i>` should have `aria-hidden` and there needs to be a text alternative.

**Comment by xi** [comment]
I noticed that django rarely sends non-200 response codes. Invalid form inputs return with 200 as does a redirect to login due to missing permissions. This code would break if the backend behavior would change.

**Comment by xi** [comment]
The ajax error handler is registered globally, so it is also triggered for requests that are triggered from the content, not just the shell. I am not sure if this is a good idea, but we can leave it for now and see how it works in practice.

File: meinberlin/static/embed.js [hunk 6]
```diff
@@ -138,6 +138,7 @@ $(document).ready(function () {
     $error.text(text)
     $error.append($close)
     $error.prependTo($('#embed-status'))
+    $close.attr('title', django.gettext('Close'))
```
Comment (vellip) [inline meinberlin/static/embed.js]: Access to title attribute information is not supported uniformly by screen readers. See: https://www.w3.org/TR/html/dom.html#the-title-attribute

File: meinberlin/static/embed.js [hunk 7]
```diff
@@ -114,6 +114,38 @@ $(document).ready(function () {
...
+        break
+    }
+
+    $error.text(text)
+    $error.append($close)
+    $error.prependTo($('#embed-status'))
+    $close.attr('title', django.gettext('Close'))
+
+    $error.on('click', removeMessage)
+    $close.on('click', removeMessage)
```
Comment (vellip) [inline meinberlin/static/embed.js]: This wouldn't be necessary anymore then.

File: meinberlin/assets/scss/components/_alert.scss [hunk 8]
```diff
@@ -8,15 +8,32 @@
...
+    color: inherit;
+}
+
+.alert--small {
+    padding: 0.5em;
+}
+
+.alert--dialog-box {
+    &:first-child {
+        padding-top: calc(16px + #{$padding * 1.2});
```
Comment (vellip) [inline meinberlin/assets/scss/components/_alert.scss]: I feel like this deserves a comment.

File: meinberlin/static/embed.js (hunk 7 above)
Comment (xi) [inline meinberlin/static/embed.js]: It is for keyboard interaction.

File: meinberlin/static/embed.js (hunk 6 above)
Comment (xi) [inline meinberlin/static/embed.js]: Both `title` and `aria-label` have pros and cons. I would like to discuss this separately.

File: meinberlin/static/embed.js (hunk 7 above)
Comment (vellip) [inline meinberlin/static/embed.js]: I tested it, worked for me without as well.

File: meinberlin/assets/scss/components/_alert.scss (hunk 8 above)
Comment (vellip) [inline meinberlin/assets/scss/components/_alert.scss]: Sorry, I meant the calculation. :X

**Pull request merged**